NickvisionMoney.Shared.Tests/CurrencyHelperTests.cs

```diff
@@ -0,0 +1,194 @@
+        RemoveSymbol(ref expected, culture);
+        FormatUnlimitedDecimals(ref expected, culture);
+        AddSymbol(ref expected, culture);
+        var result = amount.ToAmountString(culture, false);
+        Assert.Equal(expected, result);
+    }
+
+    [Theory]
+    [MemberData(nameof(GetSampleDataWithRealCultures))]
+    public void ToAmountString_RealCulturesShouldWorkWithNativeDigits(CultureInfo culture, decimal amount)
+    {
+        var expected = amount.ToString("C", culture);
+        expected =  expected
+            .Replace("0", culture.NumberFormat.NativeDigits[0])
+            .Replace("1", culture.NumberFormat.NativeDigits[1])
+            .Replace("2", culture.NumberFormat.NativeDigits[2])
+            .Replace("3", culture.NumberFormat.NativeDigits[3])
+            .Replace("4", culture.NumberFormat.NativeDigits[4])
+            .Replace("5", culture.NumberFormat.NativeDigits[5])
+            .Replace("6", culture.NumberFormat.NativeDigits[6])
+            .Replace("7", culture.NumberFormat.NativeDigits[7])
+            .Replace("8", culture.NumberFormat.NativeDigits[8])
+            .Replace("9", culture.NumberFormat.NativeDigits[9]);
+        var result = amount.ToAmountString(culture, true);
+        Assert.Equal(expected, result);
+    }
+    
+    [Theory]
+    [MemberData(nameof(GetSampleDataWithRealCultures))]
+    public void ToAmountString_RealCulturesShouldWorkWithoutCurrencySymbol(CultureInfo culture, decimal amount)
+    {
+        var expected = amount.ToString("C", culture);
+        if(culture.NumberFormat.CurrencyDecimalSeparator != culture.NumberFormat.CurrencySymbol)
+        { 
+            expected = expected.Remove(expected.IndexOf(culture.NumberFormat.CurrencySymbol), culture.NumberFormat.CurrencySymbol.Length).Trim();
+        }
+        var result = amount.ToAmountString(culture, false, false);
+        Assert.Equal(expected, result);
+    }
+    
+    [Theory]
+    [MemberData(nameof(GetSampleDataWithCustomCultures))]
+    public void ToAmountString_CustomCulturesShouldWorkWithoutCurrencySymbol(CultureInfo culture, decimal amount)
+    {
+        var expected = amount.ToString("C", culture);
+        RemoveSymbol(ref expected, culture);
+        FormatUnlimitedDecimals(ref expected, culture);
+        var result = amount.ToAmountString(culture, false, false);
+        Assert.Equal(expected, result);
+    }
+    
+    [Theory]
+    [MemberData(nameof(GetSampleDataWithRealCultures))]
+    public void ToAmountString_AllCulturesShouldWorkWithOverwriteDecimal(CultureInfo culture, decimal amount)
+    {
+        //Arrange
+        var expected = amount.ToString("C6", culture).Replace("\u200b", "").Trim();
+        if (culture.Name is "kea-CV" or "pt-CV")
+        {
+            expected = expected.TrimEnd('0');
+            if (expected.EndsWith(culture.NumberFormat.CurrencySymbol))
+                expected += "00"; // Make the currency symbol be in the center of the string
+        }
```

Comment:
Not sure what is supposed to be done here 🤔 Is it ok to leave the currency at the end? If not how many zeros should be added

Reply:
For context: this is in a special condition for the two cultures that use $ instead of .

Reply:
A lot of cultures use symbols other than `.` as decimal separator, I don't understand why do you see a problem with `$`, it should be treated like any other decimal separator. As I understand all the mess is because of invisible currency symbol. 🤔

Reply:
Nono. $ is not just the decimal separator here, it is the currency symbol itself. So if we have unlimited decimals and the input is 2$00 according to our previous logic we trim all zeros and remove the separator. But we can't remove the separator because it is the currency symbol too, so 2$ it is. Now another problem, the $ is no  longer at the middle but instead at the end. What i did here is add back 00 in this case and make it 2$00. Not sure if this should be the expected behavior.

Reply:
Wow, that's complex, sorry, I think I understand now, gonna undo my commit and see what can be done